packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -293,6 +293,8 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate {
 
     let rect = new Rect(x, y, width - x, rectHeight);
     let layoutInfo = new LayoutInfo(node.type, node.key, rect);
+    // allow overflow so the focus ring/selection ring can extend outside to overlap with the adjacent items borders
+    layoutInfo.allowOverflow = true;
```

Comment:
Just so we are aware, this will apply to ListBox, Menu, etc. as well. Is that ok?

Reply:
I do not see a problem with it. None of them use rowHeight instead of estimatedRowHeight from what I can tell. Those would be the ones to look out for.
I'll add to the test instructions to add a quick test of those other components though, that's a good call out.